tests/test_peak_properties.py

```diff
@@ -0,0 +1,98 @@
+from hypothesis import given, strategies, example, settings
+import tempfile
+
+
+@settings(max_examples=100, deadline=None)
+@given(
+    # number of peaks
+    strategies.integers(min_value=0, max_value=20),
+    # length of the data field in the peaks
+    strategies.integers(min_value=2, max_value=20),
+)
+def test_index_of_fraction(peak_length, data_length):
+    """
+    Test strax.index_of_fraction
+    """
+    # Setup the peaks
+    dtype = [(('Start time since unix epoch [ns]', 'time'), np.int64),
+             (('Exclusive end time since unix epoch [ns]', 'endtime'), np.int64),
+             (('length of p', 'length'), np.int16),
+             (('area of p', 'area'), np.float64),
+             (('data of p', 'data'), (np.float64, data_length))]
+    peaks = np.zeros(peak_length, dtype=dtype)
+    peaks['time'] = np.arange(peak_length)
+    peaks['endtime'] = peaks['time'] + 1
+
+    # Fill the peaks with random length data
+    for p in peaks:
+        length = np.random.randint(0, data_length)
+        p['length'] = length
+        wf = np.random.random(size=data_length)
+        p['data'] = wf
+
+        for i in range(length, len(wf)):
+            # Set all the out of length data to zero
+            wf[i] = 0.
+
+        if len(peaks):
+            # Compute sum area
+            peaks['area'] = np.sum(peaks['data'], axis=1)
+
+    fraction_desired = np.random.random(size=peak_length)
+    res = strax.index_of_fraction(peaks, fraction_desired)
+    assert len(res) == len(peaks), "Lost peaks"
+    if len(peaks):
+        assert np.max(res) <= data_length, "Index returned out of bound"
+
+
+@settings(max_examples=100, deadline=None)
+@given(
+    # number of peaks
+    strategies.integers(min_value=0, max_value=20),
+    # length of the data field in the peaks
+    strategies.integers(min_value=2, max_value=20),
+    # Number of widths to compute
+    strategies.integers(min_value=2, max_value=10),
+)
+def test_compute_widths(peak_length, data_length, n_widths):
+    """
+    Test strax.compute_widths
+    """
+    dtype = [(('Start time since unix epoch [ns]', 'time'), np.int64),
+             (('dt in ns', 'dt'), np.int64),
+             (('length of p', 'length'), np.int16),
+             (('area of p', 'area'), np.float64),
+             (('data of p', 'data'), (np.float64, data_length)),
+             (('width of p', 'width'), (np.float64, n_widths)),
+             (('area_decile_from_midpoint of p', 'area_decile_from_midpoint'), (np.float64, n_widths))]
+    peaks = np.zeros(peak_length, dtype=dtype)
+    dt = 1
+    peaks['time'] = np.arange(peak_length)*dt
+    peaks['dt'][:] = dt
+
+    # Fill the peaks with random length data
+    for p in peaks:
+        length = np.random.randint(0, data_length)
+        p['length'] = length
+        wf = np.random.random(size=data_length)
+        p['data'] = wf
+
+        for i in range(length, len(wf)):
+            # Set all the out of length data to zero
+            wf[i] = 0.

```

Comment:
Not impossible but I like the fact that the first test does test something whereon the second test is build.

Reply:
But this would be also the case if you join them no? But you would just initialize everything once and have less copy-paste code.

Reply:
OK, will change the setup of the random peaks to a different function, you are right that there is some unnecessary duplication. My reasoning is that I don't mind that much if the tests are sometimes a little rough/slightly duplicate, it's already tedious enough to write them while being quite unrewarding ;).

> But this would be also the case if you join them no

So the failure would be a little clearer because you split it over multiple functions with the name of what is being tested.

Reply:
>  it's already tedious enough to write them while being quite unrewarding ;). 

Yes I know :D I feel with you. But it is our task to gently bully one another for a cleaner code.

> OK, will change the setup of the random peaks to a different function, 
> So the failure would be a little clearer because you split it over multiple functions with the name of what is being tested.

Okay nice, in this way we get both.